addons/mail/static/src/core/message_model.js

```diff
@@ -130,6 +134,10 @@ export class Message {
         return this.needaction_partner_ids.includes(this._store.user?.id);
     }
 
+    get isPinnable() {
+        return this._store.self?.type === "partner";
+    }
```

Comment:
I would just remove it and simply but immediate condition in the template as `store.user` being truthy